pool/database/database.ml

```diff
@@ -115,21 +126,26 @@ end
 type event =
   | Added of Pool_database.t
   | Migrated of Pool_database.Label.t
+  | InitializedGuard of Pool_database.Label.t
 [@@deriving eq, show]
 
 let handle_event _ : event -> unit Lwt.t = function
-  | Added pool ->
-    let open Pool_database in
-    add_pool pool;
-    let%lwt () = Guard.Persistence.start ~ctx:(to_ctx pool.label) () in
-    Lwt.return_unit
+  (* Is this event still needed? *)
+  | Added pool -> Pool_database.add_pool pool |> Lwt.return
   | Migrated label ->
+    let () =
+      Logs.info (fun m -> m "Migrating: %s" Pool_database.(Label.show label))
+    in
     let%lwt () =
       match Pool_database.is_root label with
       | true -> Migration.Root.run ()
       | false -> Migration.Tenant.run [ label ] ()
     in
     Lwt.return_unit
+    (* Added: This has to run after the migrations, Should it be a separate
```

Comment:
It's the only use case at the moment. Let's add it directly to the "Migrated" event